daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/CommandPreprocessorSpec.scala

```diff
@@ -51,13 +71,120 @@ class CommandPreprocessorSpec extends AnyWordSpec with Matchers with TableDriven
+        "Transfer",
+        ValueRecord("", ImmArray("content" -> ValueList(FrontStack(ValueParty("Clara"))))),
+      )
+      val validFetch = FetchCommand(
+        "Mod:Record",
+        newCid,
+      )
+      val validFetchByKey = FetchByKeyCommand(
+        "Mod:Record",
+        valueParties,
+      )
+      val validLookup = LookupByKeyCommand(
+        "Mod:Record",
+        valueParties,
+      )
+      val negativeTestCases = Table[Command](
+        "command",
+        validCreate,
+        validExe,
+        validExeByKey,
+        validCreateAndExe,
+        validFetch,
+        validFetchByKey,
+        validLookup,
+      )
+
+      val positiveTestCases = Table[Command, ResultOfATypeInvocation[_]](
+        ("command", "error"),
+        validCreate.copy(templateId = "Mod:Undefined") ->
+          a[Error.Preprocessing.Lookup],
+        validCreate.copy(argument = ValueRecord("", ImmArray("content" -> ValueInt64(42)))) ->
+          a[Error.Preprocessing.TypeMismatch],
+        validExe.copy(templateId = "Mod:Undefined") ->
+          a[Error.Preprocessing.Lookup],
+        validExe.copy(choiceId = "Undefined") ->
+          a[Error.Preprocessing.Lookup],
+        validExe.copy(argument = ValueRecord("", ImmArray("content" -> ValueInt64(42)))) ->
+          a[Error.Preprocessing.TypeMismatch],
+        validExeByKey.copy(templateId = "Mod:Undefined") ->
+          a[Error.Preprocessing.Lookup],
+        validExeByKey.copy(contractKey = ValueList(FrontStack(ValueInt64(42)))) ->
+          a[Error.Preprocessing.TypeMismatch],
+        validExeByKey.copy(choiceId = "Undefined") ->
+          a[Error.Preprocessing.Lookup],
+        validExeByKey.copy(argument = ValueRecord("", ImmArray("content" -> ValueInt64(42)))) ->
+          a[Error.Preprocessing.TypeMismatch],
+        validCreateAndExe.copy(templateId = "Mod:Undefined") ->
+          a[Error.Preprocessing.Lookup],
+        validCreateAndExe.copy(createArgument =
+          ValueRecord("", ImmArray("content" -> ValueInt64(42)))
+        ) ->
+          a[Error.Preprocessing.TypeMismatch],
+        validCreateAndExe.copy(choiceId = "Undefined") ->
+          a[Error.Preprocessing.Lookup],
+        validCreateAndExe.copy(choiceArgument =
+          ValueRecord("", ImmArray("content" -> ValueInt64(42)))
+        ) ->
+          a[Error.Preprocessing.TypeMismatch],
+        validFetch.copy(templateId = "Mod:Undefined") ->
+          a[Error.Preprocessing.Lookup],
+        validFetchByKey.copy(templateId = "Mod:Undefined") ->
+          a[Error.Preprocessing.Lookup],
+        validFetchByKey.copy(key = ValueList(FrontStack(ValueInt64(42)))) ->
+          a[Error.Preprocessing.TypeMismatch],
+        validLookup.copy(templateId = "Mod:Undefined") ->
+          a[Error.Preprocessing.Lookup],
+        validLookup.copy(contractKey = ValueList(FrontStack(ValueInt64(42)))) ->
+          a[Error.Preprocessing.TypeMismatch],
+      )
+
+      forEvery(negativeTestCases) { command =>
```

Comment:
I’m confused why is negative = success? I would have expected the two to be flipped.

Reply:
I renamed to `noErrorTestCases` /`errorTestCases`